File: 1470_Shuffle_The_Array/Solution2.java
class Solution2 {
    public int[] shuffle(int[] nums, int n) {
        

        for (int i  = 0 ; i<n;i++){
            nums[i+n] = nums[i]*1001+nums[i+n];
        }
        for (int i = 0; i<n; i++){
            nums[2*i] = nums[i+n]/1001;
            nums[2*i+1] = nums[i+n]%1001;
        }
        return nums;

        
    }
}

File: 268. Missing Number/Solution2.cpp
class Solution2 {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        sort(begin(nums),end(nums));
        int ans = n;
        int left = 0;
        int right = n-1;
        while(left<=right){
            int mid = left + (right-left)/2;
            if(nums[mid] > mid){
                ans = mid;
                right = mid-1;
            }else{
                left = mid+1;
            }
        }
        return ans;
    }
};